jersey-servers/src/test/java/com/palantir/remoting3/servers/jersey/RequestLimitingInterceptorTest.java

```diff
@@ -0,0 +1,109 @@
+
+import io.dropwizard.Application;
+import io.dropwizard.Configuration;
+import io.dropwizard.setup.Environment;
+import io.dropwizard.testing.junit.DropwizardAppRule;
+import java.io.InputStream;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import org.glassfish.jersey.client.JerseyClientBuilder;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+public final class RequestLimitingInterceptorTest {
+
+    @ClassRule
+    public static final DropwizardAppRule<Configuration> APP =
+            new DropwizardAppRule<>(RequestLimitingTestServer.class, "src/test/resources/test-server.yml");
+
+    private WebTarget target;
+
+    @Before
+    public void before() {
+        String endpointUri = "http://localhost:" + APP.getLocalPort();
+        JerseyClientBuilder builder = new JerseyClientBuilder();
+        Client client = builder.build();
+        target = client.target(endpointUri);
+    }
+
+    @Test
+    public void testUnderLimit() {
+        Response response = target.path("/limit").request().post(Entity.json("short"));
+        assertThat(response.getStatus()).isEqualTo(204);
+    }
+
+    @Test
+    public void testOverLimit() {
+        Response response = target.path("/limit").request().post(Entity.json("over the limit"));
+        assertThat(response.getStatus()).isEqualTo(400);
+    }
+
+    @Test
+    public void testUnderLimit_streaming() {
+        Response response = target.path("/streaming").request().post(Entity.json("short"));
+        assertThat(response.getStatus()).isEqualTo(204);
+    }
+
+    @Test
+    public void testOverLimit_streaming() {
+        Response response = target.path("/streaming").request().post(Entity.json("over the limit"));
+        assertThat(response.getStatus()).isEqualTo(204);
+    }
+
+    public static class RequestLimitingTestServer extends Application<Configuration> {
+        @Override
+        public final void run(Configuration config, final Environment env) {
+            env.jersey().register(new RequestLimitingInterceptor(10));
+            env.jersey().register(new RequestLimitingTestResource());
+            env.jersey().register(HttpRemotingJerseyFeature.INSTANCE);
+        }
+    }
+
+    public static final class RequestLimitingTestResource implements RequestLimitingTestService {
+        @Override
+        public void limited(String string) {}
+
+        @Override
+        public void streaming(InputStream stream) {}
+    }
+
+    @Path("/")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public interface RequestLimitingTestService {
+        @POST
+        @Path("/limit")
+        void limited(String string);
+
+        @POST
```

Comment:
you may need to add a consumes annotation that says this is text, so jackson doesn't try to do anything with it